sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -67,6 +74,9 @@ object CollateExpressionBuilder extends ExpressionBuilder {
  */
 case class Collate(child: Expression, collationName: String)
   extends UnaryExpression with ExpectsInputTypes {
+  if (!SQLConf.get.collationEnabled) {
```

Comment:
`Expression` is private API. The only way end-user can use this expression is through SQL function lookup, so the check in `CollateExpressionBuilder` should be sufficient.